float32 wfl
float32 wfr
float32 wrl
float32 wrr
time timestamp